api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -264,6 +285,26 @@ public void setLocation(Location location) {
 	public Set<Obs> getAllObs() {
 		return getAllObs(false);
 	}
+
+	/**
+	 * Returns all encounter obs as a flatten list(Set) of obs.
+	 * @param includeVoided indicates whether or not to include voided obs
+	 * @return a Set of all encounter' Obs
+	 */
+	public Set<Obs> getAllFlattenObs(boolean includeVoided) {
+
+		Set<Obs> ret = new LinkedHashSet<>();
+
+		if (this.obs != null) {
+			for (Obs o : this.obs) {
+				if (includeVoided || (!includeVoided && !o.getVoided())) {
+					ret.add(o);
```

Comment:
Per my comments above, I think we may want to add the parent obs in the above method instead of here, to make sure we get nested parents?  Maybe the above method would be called "getFlattenedObsHelper" since it would no longer be dealing just with leaves?

Reply:
Thank you @mogoodrich . Those were great catches and suggestions. I just pushed an update that addresses all your points above. Thanks!